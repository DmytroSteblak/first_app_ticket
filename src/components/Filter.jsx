import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { Formik } from 'formik';
import{ Col, Form, FormCheck } from 'react-bootstrap';

import {
    allTickets,
    oneStopping, threeStopping,
    twoStopping,
    withoutStopped
} from '../redux/ducks/TicketsReducer';
import { NavB } from '../styled/Transplants-styled';

const Filter = () => {
    let history = useHistory();
    const dispatch = useDispatch();
    const [state, setState] = useState({
        all: false,
        no: false,
        one: false,
        two: false,
        three: false,
    });
    const { pathname } = history.location;
    useEffect(() =>{
        setState({
            [pathname.slice(1)]: true,
        });
    },[]);
    useEffect(() =>{
        setState(prevState => ({
            ...prevState, all: true
        }));
    },[]);

    const handleClickFilter = (e) =>{
        console.log(e)
        history.push(`/${e.target.defaultValue}`);
        setState({
        [e.target.defaultValue]: e.target.checked,
        });
    if (e.target.defaultValue === 'one') {
        dispatch(oneStopping());
        }
    if (e.target.defaultValue === 'two') {
        dispatch(twoStopping());
        }
    if (e.target.defaultValue === 'three') {
        dispatch(threeStopping());
        }
    if (e.target.defaultValue === 'all') {
        dispatch(allTickets());
        }
    if (e.target.defaultValue === 'no') {
        dispatch(withoutStopped());
        }
    };

    return (
        <Formik
            initialValues={{
                checked: []
            }}
            onSubmit={values => {
                console.log(values);
            }}
        >
            {() => (
                <Col lg="4" md="12" xs="12">
                    <Form.Group className="bg-white p-4 shadow 1">
                        <NavB>
                            <h6 className="mb-4">количество пересадок</h6>
                            <div>
                                <FormCheck className="check_itm">
                                    <FormCheck.Label>Все
                                        <FormCheck.Input
                                            onChange={handleClickFilter}
                                            checked={state.all}
                                            name="checked"
                                            value="all"
                                            className="mb-3"
                                        />
                                    </FormCheck.Label>
                                </FormCheck>
                                <FormCheck className="check_itm">
                                    <FormCheck.Label>Без пересадок
                                        <FormCheck.Input
                                            onChange={handleClickFilter}
                                            checked={state.no}
                                            name="checked"
                                            value="no"
                                            className="mb-3"
                                        />
                                    </FormCheck.Label>
                                </FormCheck>
                                <FormCheck className="check_itm">
                                    <FormCheck.Label>1 пересадка
                                        <FormCheck.Input
                                            onChange={handleClickFilter}
                                            checked={state.one}
                                            name="checked"
                                            value="one"
                                            className="mb-3"
                                        />
                                    </FormCheck.Label>
                                </FormCheck>
                                <FormCheck className="check_itm">
                                    <FormCheck.Label>2 пересадки
                                        <FormCheck.Input
                                            onChange={handleClickFilter}
                                            checked={state.two}
                                            name="checked"
                                            value="two"
                                            className="mb-3"
                                        />
                                    </FormCheck.Label>
                                </FormCheck>
                                <FormCheck className="check_itm">
                                    <FormCheck.Label>3 пересадки
                                        <FormCheck.Input
                                            onChange={handleClickFilter}
                                            checked={state.three}
                                            name="checked"
                                            value="three"
                                            className="mb-4"
                                        />
                                    </FormCheck.Label>
                                </FormCheck>
                            </div>
                        </NavB>
                    </Form.Group>
                </Col>
            )}
        </Formik>
    );
};

export default Filter;