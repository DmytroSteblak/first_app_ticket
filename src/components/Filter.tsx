import React, { ChangeEvent, useEffect, useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { Formik } from 'formik';
import{ Col, Form, FormCheck } from 'react-bootstrap';

import { allTickets, filterStopped } from '../redux/ducks/TicketsReducer';
import { Categories } from '../types';
import { tickets } from '../different';
import { NavB } from '../styled/Transplants-styled';

const Filter = () => {
    const history = useHistory();
    const dispatch = useDispatch();
    const [currentCategory, setCurrentCategory] = useState<Categories>(Categories.All);
    const { pathname } = useLocation();

    useEffect(() => {
        if (pathname === "/") {
            setCurrentCategory(Categories.All);
        } else {
            setCurrentCategory(pathname.slice(1) as Categories);
        }
    },[pathname]);

    const updateStopped = (e: ChangeEvent<HTMLInputElement>, id: number, value: string) => {
        history.push(`/${e.target.defaultValue}`);
        if (e.target.defaultValue === value) {
            dispatch(filterStopped(id));
        }
        if (e.target.defaultValue === 'all') {
            dispatch(allTickets());
        }
    };

    return (
        <Formik
            initialValues={{
                checked: []
            }}
            onSubmit={values => {
                console.log(values);
            }}
        >
            {() => (
                <Col lg="4" md="12" xs="12">
                    <Form.Group className="bg-white p-4 shadow 1">
                        <NavB>
                            <h6 className="mb-4">количество пересадок</h6>
                            <div>
                                {tickets.map((itm) => (
                                    <FormCheck className="check_itm" key={itm.id}>
                                        <FormCheck.Label>{itm.label}
                                            <FormCheck.Input
                                                onChange={(event: ChangeEvent<HTMLInputElement>) => updateStopped(event, itm.id, itm.value)}
                                                checked={itm.value === currentCategory}
                                                name={itm.name}
                                                value={itm.value}
                                                className="mb-3"
                                            />
                                        </FormCheck.Label>
                                    </FormCheck>
                                ))}
                            </div>
                        </NavB>
                    </Form.Group>
                </Col>
            )}
        </Formik>
    );
};

export default Filter;
