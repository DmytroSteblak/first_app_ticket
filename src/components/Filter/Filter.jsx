import React from "react";
import {ButtonComp} from "../Button/Button";
import {FilterRow} from "../../styled/Filter-component";




const Filter = (props) => (
    <FilterRow {...props}>
        <ButtonComp router={"cheap"}>
            самый дешевый
        </ButtonComp>
        <ButtonComp router={"fast"}>
            самый быстрый
        </ButtonComp>
    </FilterRow>


)

export default Filter;