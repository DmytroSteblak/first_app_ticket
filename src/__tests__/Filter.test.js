import React from 'react';
import { render } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';

import Filter from '../components/Filter';

const mockUseLocationValue = {
    pathname: '/testRoute',
    search: '',
    hash: '',
    state: null
}

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({
        useLocation: jest.fn().mockImplementation(() => {
            return mockUseLocationValue;
        })
    })
}));

describe('Testing component <Filter/>', () => {
    const initialState = { output: 10 };
    const mockStore = configureStore();
    let store;
    let component;

    beforeAll(() => {
        store = mockStore(initialState);
        component = render(
            <Provider store={store}>
                <Filter />
            </Provider>
        );
    });

    it('should Search have be rendered correctly', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

});
