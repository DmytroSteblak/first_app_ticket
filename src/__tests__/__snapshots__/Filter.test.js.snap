// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Filter/> should Search have be rendered correctly 1`] = `
<div
  class="col-lg-4 col-md-12 col-12"
>
  <div
    class="bg-white p-4 shadow 1 form-group"
  >
    <div
      class="sc-gtsrHT jlcxdi"
    >
      <h6
        class="mb-4"
      >
        количество пересадок
      </h6>
      <div>
        <div
          class="check_itm form-check"
        >
          <label
            class="form-check-label"
          >
            Все
            <input
              checked=""
              class="mb-3 form-check-input"
              name="checked"
              type="checkbox"
              value="all"
            />
          </label>
        </div>
        <div
          class="check_itm form-check"
        >
          <label
            class="form-check-label"
          >
            Без пересадок
            <input
              class="mb-3 form-check-input"
              name="checked"
              type="checkbox"
              value="no"
            />
          </label>
        </div>
        <div
          class="check_itm form-check"
        >
          <label
            class="form-check-label"
          >
            1 пересадка
            <input
              class="mb-3 form-check-input"
              name="checked"
              type="checkbox"
              value="one"
            />
          </label>
        </div>
        <div
          class="check_itm form-check"
        >
          <label
            class="form-check-label"
          >
            2 пересадки
            <input
              class="mb-3 form-check-input"
              name="checked"
              type="checkbox"
              value="two"
            />
          </label>
        </div>
        <div
          class="check_itm form-check"
        >
          <label
            class="form-check-label"
          >
            3 пересадки
            <input
              class="mb-3 form-check-input"
              name="checked"
              type="checkbox"
              value="three"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
